app/streaming/video/videoenhancement.cpp

```diff
@@ -0,0 +1,318 @@
+        index++;
+
+        if (sta != ERROR_SUCCESS)
+            continue;
+
+        std::wstring subkey(SUBKEY);
+        subkey.append(L"\\");
+        subkey.append(keyName);
+        subkey.append(L"\\");
+        subkey.append(L"0000");
+        DWORD lg;
+
+        wchar_t desc[128] = {};
+        lg = sizeof(desc) / sizeof(wchar_t);
+        if (ERROR_SUCCESS != RegGetValueW(HKEY_LOCAL_MACHINE, subkey.c_str(), L"DriverDesc",
+                                          RRF_RT_REG_SZ, nullptr, desc, &lg))
+            continue;
+
+        std::wstring s_desc(desc);
+        if (s_desc != m_GPUname)
+            continue;
+
+        // Driver of interest found, we read version
+        wchar_t charVersion[64] = {};
+        lg = sizeof(charVersion) / sizeof(wchar_t);
+        if (ERROR_SUCCESS != RegGetValueW(HKEY_LOCAL_MACHINE, subkey.c_str(), L"DriverVersion",
+                                          RRF_RT_REG_SZ, nullptr, charVersion, &lg))
+            continue;
+
+        std::wstring strVersion(charVersion);
+
+        // Convert driver store version to Nvidia version
+        if (isVendorNVIDIA()) // NVIDIA
+        {
+            strVersion = std::regex_replace(strVersion, std::wregex(L"\\."), L"");
+            m_DriverVersion = std::stoi(strVersion.substr(strVersion.length() - 5, 5));
+        }
+        else // AMD/Intel/WDDM
+        {
+            m_DriverVersion = std::stoi(strVersion.substr(0, strVersion.find('.')));
+        }
+    } while (sta == ERROR_SUCCESS);
+    RegCloseKey(hKey);
+
+#endif
+
+    return m_DriverVersion;
+}
+
+/**
+ * \brief Check if the vendor is AMD
+ *
+ * \return bool Returns true is the vendor is AMD
+ */
+bool VideoEnhancement::isVendorAMD(){
+    return m_VendorId == VENDOR_ID_AMD;
+}
+
+/**
+ * \brief Check if the vendor is Intel
+ *
+ * \return bool Returns true is the vendor is Intel
+ */
+bool VideoEnhancement::isVendorIntel(){
+    return m_VendorId == VENDOR_ID_INTEL;
+}
+
+/**
+ * \brief Check if the vendor is NVIDIA
+ *
+ * \return bool Returns true is the vendor is NVIDIA
+ */
+bool VideoEnhancement::isVendorNVIDIA(){
+    return m_VendorId == VENDOR_ID_NVIDIA;
+}
+
+/**
+ * \brief Check the Video Super-Resolution capability
+ *
+ * Check if the GPU used is capable of providing VSR feature according to its serie or driver version
+ *
+ * \return bool Returns true if the VSR feature is available
+ */
+bool VideoEnhancement::isVSRcapable(){
+    if(isVendorAMD()){
+        // [TODO] To be done once AMD provides the VSR solution
+        // Driver > 24 && RX 7000+
+    } else if(isVendorIntel()){
+        // All CPU with iGPU (Gen 10th), or dedicated GPU, are capable
+        if(m_DriverVersion >= MIN_VSR_DRIVER_VERSION_INTEL){
+            return true;
+        }
+    } else if(isVendorNVIDIA()){
+        // RTX VSR v1.5 supports NVIDIA RTX Series 20 starting from the Windows drive 545.84 (Oct 17, 2023).
+        if(
+            m_GPUname.find(L"RTX") != std::wstring::npos
```

Comment:
We shouldn't be examining the name to determine this capability (some non-consumer-branded GPUs may not have "RTX" in the name). We should use a proper API for that.

Reply:
Done
I simplified a lot VideoEnhancement.cpp by removing all Driver checks, I directly try in d3d11va.cpp the features if they can be enabled. VideoEnhancement now acts more like global variables related to Video enhancement, no logic implemented inside the class.
_Note: I havn't yet implement AMD VSR, so VideoEnhancment class may evolve depending on the needs._

Reply:
I added AMD VSR feature and it was not necessary to touch VideoEnhancement class.